### PR TITLE
testing input equallity

### PR DESCRIPTION
**Pull request by setten**

### CODE REVIEW

File: abipy/core/testing.py [hunk 1]
```diff
@@ -118,12 +120,104 @@ def has_mongodb(host='localhost', port=27017, name='mongodb_test', username=None
...
+        flat_var = []
+        if isinstance(o, tree_types):
+            for value in o:
+                for sub_value in flatten_var(value, tree_types):
+                    flat_var.append(sub_value)
+        else:
+            flat_var.append(o)
+        return flat_var
+
+    with open(ref_file) as fp:
```
Comment (gmatteo) [inline abipy/core/testing.py]: There's a problem with the pseudos more specifically with the value of pseudo.filepath stored 
in json file that is actually the absolute path used on your machine. See https://travis-ci.org/abinit/abipy/jobs/214653916

You have to preprocess the dictionary returned by json, in particular the entries 
with the pseudopotential paths: extract the `pseudo_basename` from the string and then use:

    pseudo_path = os.path.join(abidata.dirpath, "pseudos", pseudo_basename)

**Comment by setten** [comment]
I’ll try to make a hack, I use a pseudo from the abipy pseudo directory but someone programmed that the file paths are made absolute ….

**Comment by setten** [comment]
this is really difficult….

**Comment by gmatteo** [comment]
ERROR: Testing g0w0_convergence_input factory single calculation.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/abinit/abipy/abipy/abio/tests/test_factories.py", line 122, in test_convergence_inputs_single
    self.assert_input_equallity(ref_file, inputs[int(t[0])][int(t[1])])
  File "/home/travis/build/abinit/abipy/abipy/core/testing.py", line 326, in assert_input_equallity
    input_equality_check(ref_file, input_to_test, rtol=rtol, atol=atol, equal_nan=equal_nan)
  File "/home/travis/build/abinit/abipy/abipy/core/testing.py", line 211, in input_equality_check
    error = error or check_var(val_list_t[k], val_list_r[k])
  File "/home/travis/build/abinit/abipy/abipy/core/testing.py", line 171, in check_var
    elif isinstance(v, (str, unicode)):
NameError: name 'unicode' is not defined


`unicode` is gone in py3k.

Replace the test with duck typing test:

   from monty.string import is_string

or:

https://github.com/oxplot/fysom/issues/1

if you prefer the `isinstance` approach

**Pull request closed without merging**